### PR TITLE
[EXEC-114] - Migrate to Junit 5

### PR DESCRIPTION
**Pull request by arturobernalg**
The idea it's migrate all the project to JUnit 5 version

https://issues.apache.org/jira/browse/EXEC-114

### CODE REVIEW

**Comment by garydgregory** [comment]
-1 for now as this project still requires Java 7. Changing the requirement to Java 8 must be done first and discussed on the mailing list. I'm all for Java 8 FWIW.

**Comment by arturobernalg** [comment]
Hi @garydgregory 
 you're right, I didn't realize that it actually requires version 1.7. How can i start the process to migrate the libs to java 1.8?

**Comment by garydgregory** [comment]
> Hi @garydgregory
> you're right, I didn't realize that it actually requires version 1.7. How can i start the process to migrate the libs to java 1.8?

You can't, each component is different, and not everyone always agrees when it is time to update to Java 8.

One component which already _is_ on Java 8 is Commons VFS, and it still has bits that are in the JUnit 3 style, like subclassing `TestCase`. Some of the VFS tests don't even run properly from Eclipse! :-( So a TODO for the community IMO would be to update VFS to be fully on JUnit 4 as a first step (one thing at a time) and then update it to JUnit 5.

**Comment by garydgregory** [comment]
@arturobernalg 
Hi,
This component is now on Java 8.

**Comment by arturobernalg** [comment]
> @arturobernalg
> Hi,
> This component is now on Java 8.

Hey @garydgregory 

I think its already done.
TY

File: pom.xml [hunk 1]
```diff
@@ -173,6 +188,14 @@
                     </ignorePathsToDelete>
                 </configuration>
             </plugin>
+            <plugin>
```
Comment (garydgregory) [inline pom.xml:191]: @arturobernalg 
Is this really needed?

Comment (arturobernalg) [reply pom.xml]: Nop. Removed.

File: pom.xml [hunk 2]
```diff
@@ -107,8 +107,23 @@
         <contributor>
             <name>Niall Pemberton</name>
         </contributor>
+        <contributor>
+            <name>Arturo Bernal</name>
+        </contributor>
     </contributors>
 
+    <dependencyManagement>
```
Comment (garydgregory) [inline pom.xml:115]: @arturobernalg 
Why? This looks like a duplication of the parent POM. What am I missing?

**Comment by arturobernalg** [comment]
> @arturobernalg
> Do you even run local builds before pushing out a PR?
> Some of these tests will obviously fail, for example, see Exec65Test where the OS checks were not updated for JUnit 5 and cause the tests to fail.

@garydgregory 
Honestly don't fallow you.

**Comment by garydgregory** [comment]
> > @arturobernalg
> > Do you even run local builds before pushing out a PR?
> > Some of these tests will obviously fail, for example, see Exec65Test where the OS checks were not updated for JUnit 5 and cause the tests to fail.
> 
> @garydgregory Honestly don't fallow you. AFAIU the test is in Junit5.
> 
> No error in my local
> 
> [WARNING] 24 warnings [INFO] ------------------------------------------------------------------------ [INFO] BUILD SUCCESS [INFO] ------------------------------------------------------------------------ [INFO] Total time: 01:43 min [INFO] Finished at: 2022-12-04T21:07:17+01:00 [INFO] ------------------------------------------------------------------------

Self explanatory: 
```
Error:  Errors: 
Error:    Exec65Test.testExec65WithSleepUsingShellScript:76 Execute The test 'testExec65WithSleepUsingShellScript()' is not possible for OS : Linux (Exit value: 0)
[INFO] 
Error:  Tests run: 104, Failures: 0, Errors: 1, Skipped: 9
```

**Comment by arturobernalg** [comment]
> > > @arturobernalg
> > > Do you even run local builds before pushing out a PR?
> > > Some of these tests will obviously fail, for example, see Exec65Test where the OS checks were not updated for JUnit 5 and cause the tests to fail.
> > 
> > 
> > @garydgregory Honestly don't fallow you. AFAIU the test is in Junit5.
> > No error in my local
> > [WARNING] 24 warnings [INFO] ------------------------------------------------------------------------ [INFO] BUILD SUCCESS [INFO] ------------------------------------------------------------------------ [INFO] Total time: 01:43 min [INFO] Finished at: 2022-12-04T21:07:17+01:00 [INFO] ------------------------------------------------------------------------
> 
> Self explanatory @arturobernalg :
> 
> ```
> Error:  Errors: 
> Error:    Exec65Test.testExec65WithSleepUsingShellScript:76 Execute The test 'testExec65WithSleepUsingShellScript()' is not possible for OS : Linux (Exit value: 0)
> [INFO] 
> Error:  Tests run: 104, Failures: 0, Errors: 1, Skipped: 9
> ```

I guest that is why there is a comment in the java docs 

```This test currently only works for Mac OS X```

File: src/test/java/org/apache/commons/exec/issues/Exec65Test.java [hunk 3]
```diff
@@ -38,7 +45,9 @@
  */
 public class Exec65Test extends AbstractExecTest {
 
-    @Test(expected = ExecuteException.class, timeout = TEST_TIMEOUT)
+    @Test
+    @Timeout(value = TEST_TIMEOUT, unit = TimeUnit.MILLISECONDS)
+    @EnabledOnOs(LINUX)
     public void testExec65WitSleepUsingSleepCommandDirectly() throws Exception {
 
         if (!OS.isFamilyUnix()) {
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/exec/issues/Exec65Test.java:53]: This check is no longer needed.

Comment (arturobernalg) [reply src/test/java/org/apache/commons/exec/issues/Exec65Test.java:53]: done

**Pull request closed without merging**